plugins/scaffolder-backend/src/service/router.ts

```diff
@@ -569,41 +574,71 @@ export async function createRouter(
+      try {
+        const credentials = await httpAuth.credentials(req);
+        await checkPermission({
+          credentials,
+          permissions: [taskReadPermission],
+          permissionService: permissions,
+        });
 
-      const [userEntityRef] = [req.query.createdBy].flat();
-      if (
-        typeof userEntityRef !== 'string' &&
-        typeof userEntityRef !== 'undefined'
-      ) {
-        throw new InputError('createdBy query parameter must be a string');
-      }
+        if (!taskBroker.list) {
+          throw new Error(
+            'TaskBroker does not support listing tasks, please implement the list method on the TaskBroker.',
+          );
+        }
 
-      if (!taskBroker.list) {
-        throw new Error(
-          'TaskBroker does not support listing tasks, please implement the list method on the TaskBroker.',
-        );
-      }
+        const createdBy = parseStringsParam(req.query.createdBy, 'createdBy');
+        const status = parseStringsParam(req.query.status, 'status');
 
-      const [statusQuery] = [req.query.status].flat();
-      if (
-        typeof statusQuery !== 'string' &&
-        typeof statusQuery !== 'undefined'
-      ) {
-        throw new InputError('status query parameter must be a string');
-      }
+        const order = parseStringsParam(req.query.order, 'order')?.map(item => {
+          const match = item.match(/^(asc|desc):(.+)$/);
+          if (!match) {
+            throw new InputError(
+              `Invalid order parameter "${item}", expected "<asc or desc>:<field name>"`,
+            );
+          }
 
-      const tasks = await taskBroker.list({
-        createdBy: userEntityRef,
-        status: statusQuery ? (statusQuery as TaskStatus) : undefined,
-      });
+          return {
+            order: match[1] as 'asc' | 'desc',
+            field: match[2],
+          };
+        });
 
-      res.status(200).json(tasks);
+        const limit = parseStringsParam(req.query.limit, 'limit')?.map(val => {
+          const ret = Number.parseInt(val, 10);
+          if (isNaN(ret)) {
+            throw new InputError(
+              `Invalid limit parameter "${val}", expected a number`,
+            );
+          }
+          return ret;
+        });
+
+        const offset = parseStringsParam(req.query.offset, 'offset')?.map(
+          val => {
+            const ret = Number.parseInt(val, 10);
+            if (isNaN(ret)) {
+              throw new InputError(
+                `Invalid offset parameter "${val}", expected a number`,
+              );
+            }
+            return ret;
+          },
+        );
+
+        const tasks = await taskBroker.list({
+          createdBy,
+          status: status ? (status as TaskStatus[]) : undefined,
+          order,
+          limit: limit ? limit[0] : undefined,
+          offset: offset ? offset[0] : undefined,
+        });
+
+        res.status(200).json(tasks);
+      } catch (e) {
+        console.error(e);
```

Comment:
I suppose you wanted to do here something else 😅

Reply:
True, the whole try is unnecessary